app/views/response/_response_analysis.html.erb

```diff
@@ -0,0 +1,415 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Convert JSON Data to HTML Table</title>
+    <style>
+        th, td, p, input, h3 {
+            font:15px 'Segoe UI';
+        }
+        table, th, td {
+            border: solid 1px #ddd;
+            border-collapse: collapse;
+            padding: 2px 3px;
+            text-align: center;
+        }
+        th {
+            font-weight:bold;
+        }
+    </style>
+</head>
+<body>
+<br>
+	<h3>
+    	Click the button to create a dynamic table using data extracted from a JSON array.
+    </h3>
+    <br>
+    <input type='button' onclick='tableFromJson()' 
+    	value='Generate review metric' />
+        
+    <p id='showData'></p>
+<br>
+    <div id='timeTaken'></div>
+
+    
+   
+</body>
+
+<script  src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+ <script>
+
+        var final_json;
+        var response_general = null;
+       
+        
+       function makeRequest(final_json,url) {
+               return new Promise(function (resolve, reject) {
+                    let xhr = new XMLHttpRequest();
+                    xhr.open('POST', url);
+                    xhr.setRequestHeader('content-type', 'application/json');
+                    xhr.onload = function () {
+                        if (this.status >= 200 && this.status < 300) {
+                            resolve(xhr.response);
+                           
+                            response_general = xhr.response;
+                            //console.log(xhr.response);
+                        } else {
+                            reject({
+                                status: this.status,
+                                statusText: xhr.statusText
+                            });
+                        }
+                    };
+                    xhr.onerror = function () {
+                        reject({
+                            status: this.status,
+                            statusText: xhr.statusText
+                        });
+                    };
+                    xhr.send(final_json);
+                });
+
+            }
+
+
+
+         async function tableFromJson() 
+        {
+            var t0 = performance.now();
+
+            var de = $("form").serialize();      
+            $("#d").text($("form").serialize());
+
+         
+            var myBooks = [Object.fromEntries(new URLSearchParams(de))]
+            console.log(myBooks[0]);
+            var j = 0;
+            for(j =0; j <1000; j++)
+            {
+                if (document.getElementById("responses_"+j+"_comments") == null)
+                {break;}
+
+            }
+            console.log("value of j");
+            console.log(j);
+
+
+
+            var obj = new Object();           
+            var jsonString = "";
+
+            var id = 0;
+            var reviews = [];
+
+           for (var i =2; i < (j); i++) {              
```

Comment:
there may be a reason we use var here, but i think let is safer. make sure this is the correct way to do it